### ISSUE TITLE
Development and production mode use different npm dependencies

### ISSUE THREAD

**Issue by Artur-** [vaadin/flow#18979]
### Description of the bug

When I start my application using `mvn spring-boot:run` I get  the following in package.json

```
"@vaadin/vaadin-lumo-styles": "24.4.0-alpha16",
"@vaadin/vaadin-material-styles": "24.4.0-alpha16",
"@vaadin/vaadin-themable-mixin": "24.4.0-alpha16",
```

When I start using `mvn spring-boot:run -Pproduction`, the dependencies are removed

### Expected behavior

package.json is always the same, no matter how the app is run or built

### Minimal reproducible example

Reproduced in copilot

### Versions

- Vaadin / Flow version: 24.4
- Java version:
- OS version:
- Browser version (if applicable):
- Application Server (if applicable):
- IDE (if applicable):

**Comment by tltv** [comment]
These dependencies are included in default development bundle via vaadin dependency. But in copilot app in production mode it's not included. 
E.g. `copilot-client-test-flow-material` module has dependency to `vaadin`. But production profile is setup to add `vaadin-core` instead of `vaadin`. If you change `vaadin-core` to `vaadin`, it should work. Or changing vaadin to vaadin-core should work also. Point is to not mix vaadin and vaadin-core dependencies for main dependencies and plugin dependencies as it seems to make dependencies disappear from the class finder.

For reference, these two classes from `flow-components` has those missing dependencies in NpmPackage annotations and if they are missing, they're not getting in package.json: [Lumo](https://github.com/vaadin/flow-components/blob/03242cbc0f13d945f7395e52fb5b3a0754382274/vaadin-lumo-theme-flow-parent/vaadin-lumo-theme-flow/src/main/java/com/vaadin/flow/theme/lumo/Lumo.java#L32) and [Material](https://github.com/vaadin/flow-components/blob/03242cbc0f13d945f7395e52fb5b3a0754382274/vaadin-material-theme-flow-parent/vaadin-material-theme-flow/src/main/java/com/vaadin/flow/theme/material/Material.java#L31). And app works still without them because same dependencies are coming from `react-components` afaik.

**Comment by Artur-** [comment]
Why do Lumo dependencies differ between `vaadin` and `vaadin-core`?

**Comment by tltv** [comment]
They have have same dependencies because both includes `vaadin-core-internal`. So having `vaadin` already and then adding `vaadin-core` adds same dependencies twice for maven to figure it out.

**Comment by Artur-** [comment]
I still don't understand the relevancy of `vaadin` vs `vaadin-core`. I think all projects exclude dev dependencies through `vaadin-core` but it should not affect Lumo in any way

**Comment by tltv** [comment]
> I still don't understand the relevancy of `vaadin` vs `vaadin-core`. I think all projects exclude dev dependencies through `vaadin-core` but it should not affect Lumo in any way

Hilla jar contains conflicting Lumo and Material classes, a copies from components, but stripped from NpmPackage annotations. Because of this, and some algorithm in class loader, one combination of dependencies read these two classes from hilla.jar and in other combination they are read from `vaadin-lumo-theme-x.jar` and `vaadin-material-theme-x.jar`. I think it should always read from the theme packages so that you'll end up getting "@vaadin/vaadin-lumo-styles", "@vaadin/vaadin-material-styles" and "@vaadin/vaadin-themable-mixin" always as long as they are in the class path.

Removing those copies form the hilla jar looks like the fix here, but I doubt it's that simple case.

**Comment by Artur-** [comment]
This https://github.com/vaadin/hilla/pull/2106 ?

**Comment by tltv** [comment]
> This [vaadin/hilla#2106](https://github.com/vaadin/hilla/pull/2106) ?

Yeah, that would do the job. After it's merged, this one can be closed as fixed.

**Comment by tltv** [comment]
Closing as vaadin/hilla#2106 is released in 24.4.0.beta3.

**Comment by Artur-** [comment]
This now works as expected with beta3 🎉
